.github/workflows/Test-Build-Push-Deploy.yml

```
@@ -0,0 +1,129 @@
+name: CI/CD
+# Trigger jobs only when 
+# - PR is created 
+# - Merge into main
+# - Merge into production
+on:
+  push:
+    branches:
+      - main
+      - production
+  pull_request:
+
+jobs:
+  
+  test-lint:
+    name: Run unit ||| e2e tests and lint code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+
+      - name: Run Tests
+        run: echo "RUNNING TEST"
+
+      - name: Lint Check
+        run: echo "RUNNING LINT //"
+
+  build-docker-image:
+    name: Build and push docker image to ECR
+    runs-on: ubuntu-latest
+    # if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/production'
```

buuil-docker-image job needs test-lint to succeed. 
Add new line needs: [test, lint, build-docker-image]
```suggestion
   needs: [test-lint]
```